client/client.go

```diff
@@ -222,24 +223,28 @@ func (c *httpClusterClient) Do(ctx context.Context, act httpAction) (*http.Respo
 
 	var resp *http.Response
 	var body []byte
+	var cerr ClusterError
 	var err error
 
 	for _, ep := range eps {
 		hc := c.clientFactory(ep)
 		resp, body, err = hc.Do(ctx, action)
 		if err != nil {
+			cerr.Errors = append(cerr.Errors, err)
 			if err == context.DeadlineExceeded || err == context.Canceled {
-				return nil, nil, err
+				return nil, nil, cerr
```

Comment:
It may be too harsh to say the cluster is not available if the user sets a too low context deadline.